### PR TITLE
[TA] Added Recognize Custom Entities Action

### PR DESCRIPTION
**Pull request by ZulaMostafa**
This PR contains the implementation of recognize custom entities action as was explained in the issue [#24054](https://github.com/azure/azure-sdk-for-net/issues/24054).

This is a draft PR since I still have to do the following: 
- Modify `AnalyzeOperationTest`
- I have to wait until the [mismatch issue](https://dev.azure.com/msazure/Cognitive%20Services/_workitems/edit/10968334/) between the swagger and the response is solved to run the tests. 
- Update The ChangeLog
- Modify some of the DocStrings

### CODE REVIEW

**Comment by ZulaMostafa** [comment]
I tried to run some of the tests and the result isn't correct because of the mismatch between the swagger and the response, it only gets the category and the other attributes is set to `null`.

**Comment by ZulaMostafa** [comment]
@maririos I don't have the option to add reviewers, can you add @kinelski?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /// a Recognize Custom Entities action in a set of documents.
+    /// Project Name and Deployment Name are required.
+    /// </summary>
+    public class RecognizeCustomEntitiesAction
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RecognizeCustomEntitiesAction"/>
+        /// class which allows callers to specify details about how to execute
+        /// a Recognize Custom Entities action in a set of documents.
+        /// Sets the <see cref="ProjectName"/> and <see cref="DeploymentName"/> properties
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs:17]: nit: missing period `.` at the end.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        /// </summary>
+        public RecognizeCustomEntitiesAction(string projectName, string deploymentName)
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs:19]: In the rest of the client we usually add `param` tags in the docstrings for each parameter in public constructors/methods. Such as in:
```cs
        /// <summary>
        /// Initializes a new instance of the <see cref="DocumentAnalysisClient"/> class.
        /// </summary>
        /// <param name="endpoint">The endpoint to use for connecting to the Form Recognizer Azure Cognitive Service.</param>
```
In this case I would just copy the descriptions of the `ProjectName` and `DeploymentName` properties.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        /// project for this action.
+        /// </summary>
+        public string ProjectName { get; }
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the
+        /// deployment for this action.
+        /// </summary>
+        public string DeploymentName { get; }
+        /// <summary>
+        /// The default value of this property is 'false'. This means, Text Analytics service logs your input text for 48 hours,
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs:36]: Just double checking, but does the swagger or the service team confirms that this property defaults to false? If I remember correctly, the default value changes depending on the API being used.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesResultCollection.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Collection of <see cref="RecognizeCustomEntitiesActionResult"/> objects corresponding
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesResultCollection.cs:12]: This is actually a collection of `RecognizeEntitiesResult` objects.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesResultCollection.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        public string DeploymentName { get; }
+
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the
+        /// project for this action.
+        /// </summary>
+        public string ProjectName { get; }
+
+        /// <summary>
+        /// Debugger Proxy class for <see cref="RecognizeCustomEntitiesResultCollectionDebugView"/>.
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesResultCollection.cs:45]: nit: this is actually the debugger proxy class for `RecognizeCustomEntitiesResultCollection`.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClientOptions.cs [hunk 6]
```diff
@@ -37,7 +37,12 @@ public enum ServiceVersion
             /// <summary>
             /// Version 3.2-preview.1
             /// </summary>
-            V3_2_Preview_1 = 3
+            V3_2_Preview_1 = 3,
+
+            /// <summary>
+            /// Version 3.2-preview.2
+            /// </summary>
+            V3_2_Preview_2 = 4
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClientOptions.cs:45]: Usually we just support the GA versions + the latest preview, so you can remove the `V3_2_Preview_1` member and keep only `V3_2_Preview_2 = 3`.

**Review by kinelski — COMMENTED** [review]
Haven't gone through tests and samples yet, but left a few comments about implementation. Looks good!

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        /// </summary>
+        public RecognizeCustomEntitiesAction(string projectName, string deploymentName)
+        {
+            ProjectName = projectName;
+            DeploymentName = deploymentName;
+        }
+
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the
+        /// project for this action.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs:27]: not sure if it is worth saying `for this action`. It is kind of confusing

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsActions.cs [hunk 8]
```diff
@@ -60,5 +60,14 @@ public TextAnalyticsActions()
         /// This property only applies for <see cref="TextAnalyticsClientOptions.ServiceVersion.V3_2_Preview_1"/> and up.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsActions.cs:60]: When you delete the `ServiceVersion.V3_2_Preview_1` I expect this docstring to be updated to `ServiceVersion.V3_2_Preview_2`

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs (hunk 3 above)
Comment (ZulaMostafa) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesAction.cs:36]: According to the description of `DisableServiceLogs` in `RecognizeEntitiesAction` the default value is false. I actually copied the description from there.

**Comment by ZulaMostafa** [comment]
I will close this PR because I made another one: [#24245](https://github.com/Azure/azure-sdk-for-net/pull/24245). The reason behind that is that this PR should've been merged into the feature branch. but it was requested to be merged into main.

**Pull request closed without merging**